packages/apputils-extension/src/shortcuts.tsx

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) Jupyter Development Team.
+ * Distributed under the terms of the Modified BSD License.
+ */
+
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { CommandRegistry } from '@lumino/commands';
+import { Selector } from '@lumino/domutils';
+import * as React from 'react';
+import { Dialog, showDialog } from '@jupyterlab/apputils';
+import { TranslationBundle } from '@jupyterlab/translation';
+
+export function displayShortcuts(
+  app: JupyterFrontEnd,
+  trans: TranslationBundle
+) {
+  const { commands } = app;
+  const elt = document.activeElement;
+
+  /**
+   * Find the distance from the target node to the first matching node.
+   *
+   * Based on Lumino private function commands.Private.targetDistance
+   * This traverses the DOM path from `elt` to the root
+   * computes the distance from `elt` to the first node which matches
+   * the CSS selector. If no match is found, `-1` is returned.
+   *
+   * It also stops traversal if the `data-lm-suppress-shortcuts` or
+   * `data-p-suppress-shortcuts` attributes are found.
+   */
+  function matchDistance(selector: string, elt: Element | null): number {
+    let targ = elt;
+    for (
+      let dist = 0;
+      targ !== null && targ !== targ.parentElement;
+      targ = targ.parentElement, ++dist
+    ) {
+      if (targ.hasAttribute('data-lm-suppress-shortcuts')) {
+        return -1;
+      }
+      /* <DEPRECATED> */
+      if (targ.hasAttribute('data-p-suppress-shortcuts')) {
+        return -1;
+      }
+      /* </DEPRECATED> */
```

Comment:
Safe to remove, we dropped `p-` prefixes in Lumino 2.0 the current implementation [here](https://github.com/jupyterlab/lumino/blob/d08b6f71828438dd7bf1ba7f59ea706a95680822/packages/commands/src/index.ts#L1595) does not include it.

```suggestion
```

Reply:
It feels to me that we should just change signature of `targetDistance` in lumino, expose it publically and reuse here. This can be done in a follow up.